### PR TITLE
Feat/encryption

### PR DESCRIPTION
**Pull request by mrigesh**

### CODE REVIEW

File: crypto/aes/aes.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	nonce := make([]byte, gcm.NonceSize())
+	// fill nonce with random data
+	if _, err = rand.Read(nonce); err != nil {
+		return nil, err
+	}
+
+	// encrypt data
+	encryptedData := gcm.Seal(nonce, nonce, data, nil)
+
+	encryptedDataString := base64.B64Encode(encryptedData)
```
Comment (bhatua) [inline crypto/aes/aes.go]: change base64.URLEncoding.EncodeToString(

File: crypto/aes/aes.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+func Decrypt(secret string, data string) (*models.Payload, error) {
+	p := models.Payload{}
+	if data == "" {
+		return nil, nil
+	}
+
+	// decode data
+	decodedData, err := base64.B64Decode(data)
```
Comment (bhatua) [inline crypto/aes/aes.go]: change to base64.URLEncoding.DecodeString

File: crypto/crypto_test.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+			want: &models.Payload{
+				Data:          "data",
+				EncryptedData: "9zE+AFpfb3PhIfdaOlPxXZAVHb3oEiTxMYcIoDuaYVs=",
+			},
+			wantErr: false,
+		},
+		{
+			name: "TestEncryptRSA - No errors",
+			args: args{
+				keyName: "TESTKEYR",
```
Comment (bhatua) [inline crypto/crypto_test.go:232]: TESTKEYR doesn't exists

File: crypto/crypto_test.go [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+				data:    []byte("data"),
+			},
+			want: &models.Payload{
+				Data:          "data",
+				EncryptedData: "9zE+AFpfb3PhIfdaOlPxXZAVHb3oEiTxMYcIoDuaYVs=",
+			},
+			wantErr: false,
+		},
+		{
+			name: "TestEncryptRSA - error",
```
Comment (bhatua) [inline crypto/crypto_test.go]: both test cases seems same and want different result

File: crypto/crypto.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+func EncryptAES(secret string, data []byte) (*models.Payload, error) {
+	return aes.Encrypt(secret, data)
+}
+
+func DecryptAES(secret string, data string) (*models.Payload, error) {
+	return aes.Decrypt(secret, data)
+}
+
```
Comment (bhatua) [inline crypto/crypto.go:25]: expose loandKeysFromFile

File: crypto/rsa/rsa.go [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+	}
+
+	decryptedData, err := keyInfo.Decrypt([]byte(encryptedData))
+	if err != nil {
+		return nil, err
+	}
+
+	return &models.Payload{
+		KeyName:       keyName,
+		KeyVersion:    keyVersion,
```
Comment (bhatua) [inline crypto/rsa/rsa.go:118]: add keyType

File: crypto/crypto_test.go (hunk 3 above)
Comment (mrigesh) [reply crypto/crypto_test.go:232]: if key doesn't exist. not returning error, but returning the same data as input

File: crypto/crypto_test.go (hunk 4 above)
Comment (mrigesh) [reply crypto/crypto_test.go]: fixed

**Review by bhatua — APPROVED** [review]

**Pull request merged**